src/classes/OPP_CampaignMember_TDTM.cls

```diff
@@ -96,34 +106,68 @@ public class OPP_CampaignMember_TDTM extends TDTM_Runnable {
-    * @description Adds a Contact Id, Campaign Id, and Status to our three lists. If the opportunity has a
+    * @description For a given list of opportunities, clones them and adds the primary opportunity contact
+    * role's contact id to the Primary_Contact__c field.
+    * @param listOpps A list of Opportunities.
+    * @return List<Opportunity> A cloned list of opportunities with the primary OCR's contact copied to the 
+    * Primary_Contact__c field.
+    ********************************************************************************************************/
+    private List<Opportunity> getOppsWithConroles(list<Opportunity> listOpps) {
+        Map<Id,Opportunity> mapResults = new Map<Id,Opportunity>();
+        //make a map of cloned opportunities so we can update the Primary_Contact__c field, which can't be written in an after insert context
+        for (Opportunity opp : listOpps) {
+            mapResults.put(opp.id, opp.clone(true,true,false,true));
+        }
+
+        for (OpportunityContactRole oppConRole : [SELECT OpportunityId, ContactId FROM OpportunityContactRole WHERE IsPrimary=TRUE AND OpportunityId IN :listOpps]) {
+            mapResults.get(oppConRole.OpportunityId).Primary_Contact__c = oppConRole.ContactId;
+        }
+        return mapResults.values();
+    }
+
+    /*******************************************************************************************************
+    * @description Checks if an opportunity's Primary_Contact_Campaign_Member_Status__c is the "omit" value.
+    * @param Opp The current opportunity.
+    * @return Boolean True if Primary_Contact_Campaign_Member_Status__c == Label.CampaignMemberStatusOmit
+    ********************************************************************************************************/
+    private Boolean isOmit(Opportunity opp) {
+        if (!string.isBlank(opp.Primary_Contact_Campaign_Member_Status__c) && opp.Primary_Contact_Campaign_Member_Status__c.equalsIgnoreCase(Label.CampaignMemberStatusOmit)) {
+            return true;
+        }
+        return false;
+    }
+
+    /*******************************************************************************************************
+    * @description Adds a Contact Id, Campaign Id, and Status to our three Lists. If the opportunity has a
     * status it is used, otherwise a default is used based on if the opportunity is closed/won or not.
     * @param Opp The current opportunity.
-    * @return void This method adds to three lists and returns nothing.
+    * @return void This method adds to three Lists and returns nothing.
     ********************************************************************************************************/
     private void addOppMemberFields(Opportunity opp) {
-        if (opp.Primary_Contact__c == null || opp.CampaignId == null)
+        if (opp.CampaignId == null || opp.Primary_Contact__c == null) {
```

Comment:
Ha, no, I was in process of changing that method but ended up reverting it backwards.